modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.cc

```diff
@@ -2100,180 +2100,329 @@ LBSSolver::ReadRestartData(const std::string& folder_name, const std::string& fi
 }
 
 void
-LBSSolver::WriteGroupsetAngularFluxes(const LBSGroupset& groupset, const std::string& file_base)
+LBSSolver::WriteAngularFluxes(const std::vector<std::vector<double>>& src,
+                              const std::string& file_base) const
 {
+  // Open the file
   std::string file_name = file_base + std::to_string(Chi::mpi.location_id) + ".data";
-
-  // Open file
   std::ofstream file(file_name,
                      std::ofstream::binary |  // binary file
                        std::ofstream::out |   // no accidental reading
-                       std::ofstream::trunc); // clear file contents when opened
+                       std::ofstream::trunc); // clear contents first
+  ChiLogicalErrorIf(not file.is_open(), "Failed to open " + file_name + ".");
+  Chi::log.Log() << "Writing angular flux to " << file_base;
+
+  // Write the header
+  const int num_bytes = 500;
```

Comment:
is 500 always enough? how was this chosen? I know it was there before, just wondering.

Reply:
Yes this is just for the fixed size `header_info` string and not for the data.